internal/bminventory/inventory.go

```diff
@@ -3149,8 +3149,13 @@ func (b *bareMetalInventory) V2DeregisterHostInternal(ctx context.Context, param
 	}
 
 	// TODO: need to check that host can be deleted from the cluster
-	b.eventsHandler.AddEvent(ctx, params.InfraEnvID, &params.HostID, models.EventSeverityInfo,
-		fmt.Sprintf("Host %s: deregistered from cluster", params.HostID.String()), time.Now())
+	infraEnv, err := common.GetInfraEnvFromDB(b.db, params.InfraEnvID)
+	var clusterID strfmt.UUID
+	if err != nil {
+		return err
+	}
+	clusterID = infraEnv.ClusterID
+	eventgen.SendHostDeregisteredClusterEvent(ctx, b.eventsHandler, clusterID, params.HostID, params.InfraEnvID)
```

Comment:
i think event should be raised regardless of InfraEnv existence or no ( if infra env does not exist, then we are deregistering dangling host - @avishayt ?).  In that case maybe we don't need to check the existence of the InfraEnv, but check the existence of the host at the beginning of the function. In case host exists, we can get all the needed values m including clusterID from it.

Reply:
Agree

Reply:
Ditto. 
I see this change got merged in #2497, so we should probably open a ticket to follow up with that.
@masayag

Also: The comment makes sense to me, but waiting for @avishayt's input here.